src/viam/ur/module/ur_arm.cpp

```diff
@@ -544,7 +544,6 @@ void URArm::trajectory_done_cb_(const control::TrajectoryResult state) {
 
 void URArm::reconfigure(const Dependencies& deps, const ResourceConfig& cfg) {
     const std::unique_lock wlock{config_mutex_};
-    check_configured_(wlock);
```

Comment:
if `configure_` throws then we have no `current_state_`, which causes the resource to fail without being able to recover

Reply:
This was by design. My understanding was that if a resource failed reconfigure, that it would be removed from the resource graph and destroyed. So the only way that something could be `reconfigure`d was if it had either 1) been created, or 2) had already been `reconfigure'd OK. Either way, there should always be a state that is being reconfigured.

So, I'm not sure about this one. But I'm OK with being wrong, as long as there is a good explanation for how the resource can end up in a "failed to configure state" such that it would see another `reconfigure` call where `check_configured_` would fail.

Reply:
I checked with @randhid, and this was probably my misunderstanding. I had believed that if you failed a `reconfigure` that the node would be removed from the resource graph. This appears not to be true: it will continue to attempt to reconfigure the resource. So, my check was too restrictive, and I agree with removing it. Especially if you have the empirical answer that it really does happen.